### PR TITLE
Dependency injection

### PR DESCRIPTION
**Pull request by Vo1z**

### CODE REVIEW

File: Core/Systems/IEcsDisposeSystem.cs [hunk 1]
```diff
@@ -2,6 +2,6 @@
 {
 	public interface IEcsDisposeSystem : IEcsSystem
 	{
-		public void OnDispose();
+		public void OnOnDispose();
```
Comment (Luke-Wojciechowski) [inline Core/Systems/IEcsDisposeSystem.cs:5]: let's just stick to OnDisponse... Creating a system (class) with method OnOnDispose seems like a misspelling and it's kinda confusing

File: Core/Systems/IEcsInitSystem.cs [hunk 2]
```diff
@@ -2,6 +2,6 @@
 {
 	public interface IEcsInitSystem : IEcsSystem
 	{
-		public void OnInit();
+		public void OnOnInit();
```
Comment (Luke-Wojciechowski) [inline Core/Systems/IEcsInitSystem.cs:5]: same here.... just let it be OnInit

File: Core/Systems/EcsSystems.cs [hunk 3]
```diff
@@ -1,65 +1,58 @@
...
 
 		[MethodImpl(MethodImplOptions.AggressiveInlining)]
 		public void FireInitSystems()
 		{
-			foreach(var initSystem in _initSystems)
-			{
-				initSystem.OnInit();
-				_world.UpdateFilters();
-			}
+			OnInitFired?.Invoke();
```
Comment (Luke-Wojciechowski) [inline Core/Systems/EcsSystems.cs:40]: last time u thought filters should be updated each time after system's execution... if not, im fine with that change.

File: Core/Systems/IEcsRunSystem.cs [hunk 4]
```diff
@@ -2,6 +2,6 @@
 {
 	public interface IEcsRunSystem : IEcsSystem
 	{
-		public void OnRun();
+		public void OnOnRun();
```
Comment (Luke-Wojciechowski) [inline Core/Systems/IEcsRunSystem.cs:5]: just OnRun

File: Core/Systems/IEcsDisposeSystem.cs (hunk 1 above)
Comment (Vo1z) [reply Core/Systems/IEcsDisposeSystem.cs:5]: oops

**Review by Luke-Wojciechowski — APPROVED** [review]

**Pull request merged**